### PR TITLE
ng_nettest: initial import of a ng_netapi test framework

### PR DESCRIPTION
**Pull request by authmillenon**
This will allow us to test `ng_netapi` modules more isolated (except MAC layers).

### CODE REVIEW

File: sys/include/net/ng_nettest.h [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+/**
+ * @brief   Test @ref NG_NETAPI_MSG_TYPE_SND command to @p pid.
+ *
+ * @details This registered the nettest thread to (@p exp_type, @p exp_demux_ctx)
+ * and checks if @p exp_pkts of @p exp_out were received from @p exp_senders.
+ * If no message was received after @reg NG_NETTEST_TIMEOUT microseconds, while
+ * there are still packets expected, the function will return NG_NETTEST_TIMED_OUT.
+ *
+ * @param[in] pid           The thread you want test the @ref NG_NETAPI_MSG_TYPE_SND
+ *                          command for.
```
Comment (phiros) [inline sys/include/net/ng_nettest.h]: I think there is a "to" missing here

File: sys/include/net/ng_nettest.h [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+/**
+ * @brief   Test @ref NG_NETAPI_MSG_TYPE_RCV command to @p pid.
+ *
+ * @details This registered the nettest thread to (@p exp_type, @p exp_demux_ctx)
+ * and checks if @p exp_pkts of @p exp_out were received from @p exp_senders.
+ * If no message was received after @reg NG_NETTEST_TIMEOUT microseconds, while
+ * there are still packets expected, the function will return NG_NETTEST_TIMED_OUT.
+ *
+ * @param[in] pid           The thread you want test the @ref NG_NETAPI_MSG_TYPE_RCV
+ *                          command for.
```
Comment (phiros) [inline sys/include/net/ng_nettest.h]: Same here ("to" missing?). Seems like a copy&paste error

File: sys/include/net/ng_nettest.h [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+/**
+ * @brief   Test @ref NG_NETAPI_MSG_TYPE_SND command to @p pid with the receiving
+ *          thread being an interface.
+ *
+ * @details This registered the nettest thread as an interface and checks ifx
+ * @p exp_pkts of @p exp_out were received from @p exp_senders. If no message
+ * was received after @ref NG_NETTEST_TIMEOUT microseconds, while there are
+ * still packets expected, the function will return NG_NETTEST_TIMED_OUT.
+ *
+ * @param[in] pid           The thread you want test to the
```
Comment (phiros) [inline sys/include/net/ng_nettest.h]: I think it should read:

The thread you want _to_ test the ... command for.

Comment (authmillenon) [inline sys/include/net/ng_nettest.h]: Gnarf, of course

Comment (authmillenon) [inline sys/include/net/ng_nettest.h]: fixed

**Comment by phiros** [comment]
Looks good to me so far (apart from the minor grammar issue in the documentation).

However, I think it would be beneficial if you could supply some simple tests using this framework. 
This would not only be useful for seeing how this thing works as a whole but also would allow us to run some additional static code tests on your implementation  (such as llvm's scan-build).

**Comment by authmillenon** [comment]
True, but I'd rather add them as seperate PRs when I find some time to write them.

**Comment by phiros** [comment]
Better late then never: ACK (after a rebase).

**Comment by authmillenon** [comment]
Squashed

**Pull request merged**